tests/commands/test_delete_project.py

```diff
@@ -0,0 +1,84 @@
+# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean CLI v1.0. Copyright 2021 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from click.testing import CliRunner
+
+from lean.commands import lean
+from lean.components.api.project_client import ProjectClient
+from tests.test_helpers import create_fake_lean_cli_directory, create_api_project
+
+
+def assert_project_exists(path: str) -> None:
+    project_dir = (Path.cwd() / path)
+
+    assert project_dir.exists()
+    assert (project_dir / "main.py").exists()
+    assert (project_dir / "research.ipynb").exists()
+
+
+def assert_project_does_not_exist(path: str) -> None:
+    project_dir = (Path.cwd() / path)
+    assert not project_dir.exists()
+
+
+def test_delete_project_locally_that_does_not_have_cloud_counterpart() -> None:
+    create_fake_lean_cli_directory()
+
+    path = "Python Project"
+    assert_project_exists(path)
+
+    with mock.patch.object(ProjectClient, 'get_all', return_value=[]) as mock_get_all,\
```

Comment:
To cover more behavior could even make the cloud api call return some other projects but not the requested one 👍